### ISSUE TITLE
no-unused-vars false positive on `async` keyword when async arrow func returns arrow func

### ISSUE THREAD

**Issue by soryy708** [eslint/eslint#12686]
**Tell us about your environment**

* **ESLint Version:** 6.5.1
* **Node Version:** 12.4.0
* **npm Version:** 6.12.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "parserOptions": {
        "ecmaVersion": 2018
    },
    "env": {
        "es6": true
    },
    "rules": {
        "no-unused-vars": "error"
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const foo = async => () => {};
```

I'm declaring an async function that returns a function.

```bash
eslint .
```

**What did you expect to happen?**
1:7   error  'foo' is assigned a value but never used  no-unused-vars

ESLint should only flag `foo` as an unused variable.

**What actually happened? Please include the actual, raw output from ESLint.**
1:7   error  'foo' is assigned a value but never used  no-unused-vars
1:13  error  'async' is defined but never used         no-unused-vars
✖ 2 problems (2 errors, 0 warnings)

ESLint flagged the keyword `async` as an unused variable.

**Are you willing to submit a pull request to fix this bug?**
No.

**Comment by soryy708** [comment]
Nevermind, I goofed up.
